hystrix-contrib/hystrix-metrics-event-stream/src/main/java/com/netflix/hystrix/contrib/sample/stream/HystrixSampleSseServlet.java

```diff
@@ -163,8 +163,17 @@ public void onNext(String sampleDataAsString) {
                 while (moreDataWillBeSent.get() && !isDestroyed) {
                     try {
                         Thread.sleep(pausePollerThreadDelayInMs);
+                        //in case stream has not started emitting yet, catch any clients which connect/disconnect before emits start
+                        writer.print("ping: \n\n");
+                        // explicitly check for client disconnect - PrintWriter does not throw exceptions
+                        if (writer.checkError()) {
+                            throw new IOException("io error");
```

Comment:
To avoid the cost of creating and throwing the exception, can you call `moreDataWillBeSent.set(false);` instead?